Update tutorials

SimonHeybrock opened the pull request:
Update and improve the two introduction tutorials that we also use for our courses. The goal of this is to be able to use identical notebooks in the course since maintaining separate variants is not sustainable in the long run.

See also https://github.com/ess-dmsc-dram/python-course-ikon/tree/master/notebooks/7_scipp.

nvaytet commented on docs/tutorials/working-with-masks.ipynb:
```diff
@@ -117,20 +129,26 @@
     "tof = data.coords['tof']\n",
-    "data.masks['prompt-pulse'] = (tof['tof',:-1] > 17500.0 * sc.units.us) & (tof['tof',1:] < 19000.0 * sc.units.us)\n",
-    "sc.plot({'before':counts, 'after':sc.sum(data, 'tof')})"
+    "data.masks['prompt-pulse'] = (tof['tof',:-1] > prompt_start) & (tof['tof',1:] < prompt_stop)\n",
```
Why are we doing the masking a second time here, if it has already been done 2 cells above?

SimonHeybrock commented on docs/tutorials/exploring-data.ipynb:
```diff
@@ -73,14 +78,28 @@
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "jp-MarkdownHeadingCollapsed": true,
```
It is supposed to collapse everything until the next markdown heading, i.e., the whole section.

SimonHeybrock commented:
Thanks, I believe I addressed your comments. The markdown heading collapse appears to be partially broken. I suggest we wait for until if upstream will fix this.

SimonHeybrock commented:
I have opened https://github.com/jupyterlab/jupyterlab/issues/11119 upstream.
